spring-integration-amqp/src/main/java/org/springframework/integration/amqp/dsl/AmqpInboundGatewaySpec.java

```diff
@@ -270,10 +272,19 @@ public AmqpInboundGatewaySpec defaultRequeueRejected(boolean defaultRequeueRejec
 		return this;
 	}
 
-	@Override
-	public Collection<Object> getComponentsToRegister() {
-		return Collections.<Object>singleton(this.listenerContainer);
+	/**
+	 * @param containerId the bean name for internal listener container instance.
```

Comment:
/* Set the bean name of the listener container if it is created internally or a non-bean
    /* object was supplied. Ignored if the container is already a bean.